RedHatInsights/turnpike
# Turnpike slo sli

SLO/SLI for turnpike:
This PR adds two sets of metrics for prometheus to scrape:
1. Auth plugin metrics.
   - Metrics exposed on `localhost:5000/metrics`
   - Metrics consist of response time and response code labeled per plugin
2. Nginx metrics.
   - Metrics exposed on `localhost:24321/metrics`
   - Metrics read from access logs of nginx which are written to a shared volume for fluentd to read
   - Metrics are labeled by method (GET, POST, etc) and status (200, 400, etc):
       - Upstream response time
       - request total duration
       - Bytes sent

To test:
- get turnpike running ([docs](https://docs.engineering.redhat.com/pages/viewpage.action?pageId=271994122)
- upon hitting https://turnpike.example.com/api/turnpike/identity/ or any other turnpike endpoint you should see logs from web_1 and fluentd_1 printing metrics
- go to the endpoints `localhost:5000/metrics` and `localhost:24321/metrics` to see the metrics being exposed for prometheus.
- if you have a local prometheus running you can configure it to scrape those endpoints
